pkg/reconciler/route/route.go

```diff
@@ -187,51 +186,20 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 
 	logger.Infof("Reconciling route: %#v", r)
 
-	// Get all services
-	existingServices, err := c.getServices(r) // This is what we use to determine whether we default or not.
-	if err != nil {
+	if err := c.updateRouteStatusURL(ctx, r); err != nil {
```

Comment:
nit: can we move this call `c.updateRouteStatusURL(ctx, r)` to right after `serviceNames, err := c.getServiceNames(ctx, r)` on line 196, so that we can pass `serviceNames.clusterLocal()` into `updateRouteStatusURL` and can avoid calling `getServiceNames` again in `updateRouteStatusURL`?

Reply:
We can, not a problem :)